sql/parser/datum.go

```diff
@@ -237,7 +237,28 @@ func (d DFloat) IsMin() bool {
 }
 
 func (d DFloat) String() string {
-	return strconv.FormatFloat(float64(d), 'g', -1, 64)
+	s := strconv.FormatFloat(float64(d), 'g', -1, 64)
+	if parsesAsInt(s) {
```

Comment:
Interesting. Then perhaps use `'f'` for the formatting flag instead of `'g'` for "whole" numbers.